### ISSUE TITLE
Support multiple destinations in make_uri and parse_uri [$350]

### ISSUE THREAD

**Issue by woodser** [haveno-dex/haveno#83]
This issue requests supporting multiple addresses and amounts in monero-wallet-rpc's [make_uri](https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#make_uri) and [parse_uri](https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#parse_uri).

This will be used to deposit 2 outputs to the Haveno wallet, one to cover the deposit amount and one to cover the trade fee, in order to allow trades to be posted and taken once funds are available.

See https://github.com/monero-project/monero/issues/7731

**Comment by benevanoff** [comment]
I am beginning work on this now

**Comment by erciccione** [comment]
Thanks @benevanoff. Please use #haveno-dev on matrix or freenode for questions and coordination :)

**Comment by woodser** [comment]
Note multi-output deposits are no longer needed with the new trade protocol, but we will honor the bounty once merged into monero-project.

**Comment by darkdrag00n** [comment]
As mentioned [here](https://github.com/monero-project/monero/issues/7731#issuecomment-1335466290), I am picking this one up.

**Comment by darkdrag00n** [comment]
@woodser I've raised https://github.com/monero-project/monero/pull/8665. Can you kindly review and also kickoff the workflows?

**Comment by NorrinRadd** [comment]
@woodser is this still open?

**Comment by woodser** [comment]
Yes this has not been completed. A PR was submitted to monero-project but never merged.

**Comment by NorrinRadd** [comment]
Ok, I will take it @woodser

**Comment by U65535F** [comment]
It's been a long time since the assignment and it hasn't completed yet? Is this still being worked on?

**Comment by woodser** [comment]
The previous solution in progress was deleted from the monero-project repo somehow, so this issue is still available for someone to complete.

**Comment by U65535F** [comment]
@woodser I am actively working on this thing. Check PR 9756 on monero code base. Check if it meets the requirements.

**Comment by U65535F** [comment]
> Linking to the PR for reference

Could you have a look at the code and see if it actually meets all of your requirements? Any suggestions would be appreciated. The mods are taking a while to respond (totally okay, it's not their job so obviously they don't fully dedicate their time in the repo).

**Comment by woodser** [comment]
A new PR is open for this: https://github.com/monero-project/monero/pull/9830

**Comment by proofline-studio** [comment]
Proofline Studio can work on a focused implementation of this bounty: wallet-independent URI generation/parsing, RPC integration, and offline fixture-based tests preserving single-recipient compatibility. Development and review would be performed by an AI agent.

The current upstream URI_SCHEME.md describes semicolon-separated lists, while monero-project/monero#9830 targets the proposed versioned, repeated-address format. Which format should a mergeable implementation follow, and would you prefer a focused replacement PR or help completing #9830?

I understand that payment of the advertised $350 bounty is in XMR, depends on upstream Monero merge, and may be shared for collaborative work. No completed implementation, reserved assignment, or human review is claimed.

**Comment by proofline-studio** [comment]
A concrete implementation is now available for review: https://github.com/monero-project/monero/pull/11353 (draft, based on current master).

It adds wallet-independent multi-recipient URI functions, additive make_uri_multi/parse_uri_multi RPC calls, grouped optional amounts/labels, exact decimal currency notation, documentation, and regression tests. The existing scalar API and CLI behavior are preserved.

20 offline C++ tests passed against the actual syntax, JSON serializer, and address-validation implementations; the complete RPC server source also passed syntax compilation. Full upstream build and live-RPC validation remain pending, and CI has been queued. The PR identifies the experimental protocol/API choices for maintainer review.

This is AI-developed and AI-reviewed work under Proofline Studio. It is not yet merged or an earned bounty.
